app/Http/Controllers/VideoController.php

```diff
@@ -0,0 +1,190 @@
+
+        } catch (Exception $e) {
+            return new BaseResponse(STATUS_CODE_BADREQUEST, STATUS_CODE_BADREQUEST, $e->getMessage() . $e->getLine() . $e->getFile() . $e);
+        }
+    }
+    
+    // Fetching Single Video | Open Video
+    public function getVideo(Request $request)
+    {
+        try {
+            $video_id = $request->id;
+
+            $video = Video::with('user')->find($video_id);
+          
+            if(!$video) {
+                return new BaseResponse(STATUS_CODE_NOTFOUND, STATUS_CODE_NOTFOUND, 'Video not found');
+            }
+
+            $mediaItem = $video->getMedia();
+            $videoUrl  = $mediaItem->first()->getUrl();
+
+            $user      = $video->user;
+
+            // Creating Response data for API
+            $response = [
+                'id'            => $video->id,
+                'title'         => $video->title,
+                'description'   => $video->description,
+                'video_url'     => $videoUrl,
+                'views_count'   => $video->views->count(),
+                'user'          => [
+                    'user_id'   =>  $user->id,
+                    'full_name' =>  $user->fullname,
+                    'image'     =>  $user->userDetails?->image,   
+                ],
+            ];
+
+            return new BaseResponse(STATUS_CODE_OK, STATUS_CODE_OK, "Video Fetched successfully", $response);
+            
+        } catch (Exception $e) {
+            DB::rollback();
+            return new BaseResponse(STATUS_CODE_BADREQUEST, STATUS_CODE_BADREQUEST, $e->getMessage() . $e->getLine() . $e->getFile() . $e);
+        }
+    }
+
+    // Current User | View All User's Videos
+    public function getCurrentUserVideos(Request $request)
+    {
+        try {
+            $user_id = $this->currentUser->id;
+
+            $videos  = Video::where('user_id', $user_id)->where('status', config('constants.video.approved'))->get();
+
+            if(!$videos) {
+                return new BaseResponse(STATUS_CODE_NOTFOUND, STATUS_CODE_NOTFOUND, 'Videos not found');
+            }
+    
+            foreach ($videos as $video) {
+                $mediaItem            = $video->getFirstMedia();
+                $video['video_url']   = $mediaItem->getUrl();
+                $video['views_count'] = $video->views->count();
+
+                unset($video['media'], $video['views']);
+            }
+
+            return new BaseResponse(STATUS_CODE_OK, STATUS_CODE_OK, 'Videos Fetched Successfully', $videos);
+
+        } catch (Exception $e) {
+            DB::rollback();
+            return new BaseResponse(STATUS_CODE_BADREQUEST, STATUS_CODE_BADREQUEST, $e->getMessage() . $e->getLine() . $e->getFile() . $e);
+        }
+    }
+
+    // Open Video to View.
+    public function viewVideo(Request $request)
+    {
+        try {
+            $video_id = $request->id;
+            $this->currentUser->views()->attach($video_id);
```

Comment:
We should improve this by adding the feature, A user cannot increase the view count, if they watch the video again